### PR TITLE
Feature/order page

### PR DESCRIPTION
**Pull request by MarkiyanSHCH**

### CODE REVIEW

File: UI/angular/src/app/components/cart/cart.component.html [hunk 1]
```diff
@@ -1,46 +1,62 @@
...
-    </div>
-    <div class="col-12 col-md-1 d-flex justify-content-between mb-2">
-      <span>
-        <span class="d-md-none fw-bold">Price: </span>
-        <span>${{dataItem.quantity*returnProduct(dataItem.productId).price}}</span>
-      </span>
-      <button class="btn btn-outline-dark d-md-none"
-              (click)="delete(i)">Delete
-      </button>
+<div class="container pt-5 pt-md-6 pb-5">
```
Comment (BodyDuk) [inline UI/angular/src/app/components/cart/cart.component.html:1]: py-5

File: UI/angular/src/app/components/cart/cart.component.html [hunk 2]
```diff
@@ -1,46 +1,62 @@
...
+  <h3 class="mb-4 mb-lg-5"
+      data-seel-text="page-header">Shopping Bag ({{cartItems.length}})</h3>
```
Comment (BodyDuk) [inline UI/angular/src/app/components/cart/cart.component.html:3]: data-seel-text="page-header"???????

File: UI/angular/src/app/components/cart/cart.component.html [hunk 3]
```diff
@@ -1,46 +1,62 @@
...
     </div>
   </div>
-</div>
 
-<div class="d-flex justify-content-end mt-2">
-  <button class="btn btn-outline-dark px-4 py-2"
-          (click)="onOrderClick()">
-    Order
-  </button>
+  <div *ngIf="cartItems.length === 0">
```
Comment (BodyDuk) [inline UI/angular/src/app/components/cart/cart.component.html:53]: Why do you need this div?

File: UI/angular/src/app/components/cart/cart.component.html [hunk 4]
```diff
@@ -1,46 +1,62 @@
...
+    <div class="row">
+      <div class="col-12 col-md-4">
+        <h5 class="mb-6 text-transform-none">Your shopping bag is empty.</h5>
+        <button routerLink="/"
+                class="btn btn-outline-dark my-2 px-4 py-2">Continue Browsing</button>
```
Comment (BodyDuk) [inline UI/angular/src/app/components/cart/cart.component.html:58]: Why do you need mb?

File: UI/angular/src/app/components/orders/order-collapse/order-collapse.component.html [hunk 5]
```diff
@@ -9,4 +10,14 @@
   <div class="col-4">{{shop.address}}</div>
   <div class="col-3">{{order.orderDate | date:'shortDate'}}</div>
   <div class="col-3">${{order.totalPrice}}</div>
+  <div id="{{order.id}}"
```
Comment (BodyDuk) [inline UI/angular/src/app/components/orders/order-collapse/order-collapse.component.html:13]: In a row, must be only col.

File: UI/angular/src/app/components/orders/order-line/order-line.component.css [hunk 6]
```diff
@@ -0,0 +1,4 @@
+.image {
+  max-width: 100%;
```
Comment (BodyDuk) [inline UI/angular/src/app/components/orders/order-line/order-line.component.css:2]: If you reuse this style, make it global.

File: UI/angular/src/app/components/orders/orders.component.ts [hunk 7]
```diff
@@ -31,7 +34,7 @@ export class OrdersComponent implements OnInit {
         .subscribe(res => this.orders = res.orders),
       shop: this._shopHttpService
         .getShops()
-        .subscribe(res => this.shops = res.shops)
+        .subscribe(res => this.shops = res.shops),
```
Comment (BodyDuk) [inline UI/angular/src/app/components/orders/orders.component.ts:37]: Redundant coma.

File: UI/angular/src/app/components/orders/orders.component.ts [hunk 8]
```diff
@@ -16,13 +17,15 @@ import { ShopHttpService } from 'src/app/services/http/shop.http.service';
...
   constructor(
     private _orderHttpService: OrderHttpService,
     private _shopHttpService: ShopHttpService
-  ) { }
+  ) {
+    this.orderLinesMap.clear();
+  }
 
   ngOnInit(): void {
     forkJoin({
```
Comment (BodyDuk) [inline UI/angular/src/app/components/orders/orders.component.ts:31]: Pleas fix forkJoin

File: UI/angular/src/app/components/orders/orders.component.ts [hunk 9]
```diff
@@ -40,8 +43,9 @@ export class OrdersComponent implements OnInit {
   }
```
Comment (BodyDuk) [inline UI/angular/src/app/components/orders/orders.component.ts:43]: Use better naming.

File: UI/angular/src/app/components/orders/orders.component.ts [hunk 10]
```diff
@@ -40,8 +43,9 @@ export class OrdersComponent implements OnInit {
...
 
   getOrderLines(orderId: number) {
-    this._orderHttpService
-      .getOrderLines(orderId)
-      .subscribe(res => this.orderLines = res.orderLines);
+    if (!this.orderLinesMap.get(orderId))
```
Comment (BodyDuk) [inline UI/angular/src/app/components/orders/orders.component.ts:46]: Try to add spinner.

File: UI/angular/src/app/components/products/detail/detail.component.ts [hunk 11]
```diff
@@ -39,6 +39,7 @@ export class DetailComponent implements OnInit {
   addCartProduct() {
     const modalRef = this._modalService.open(CartModuleComponent, { centered: true });
     const module = <CartModuleComponent>modalRef.componentInstance;
+    console.log(this.product)
```
Comment (BodyDuk) [inline UI/angular/src/app/components/products/detail/detail.component.ts:42]: Delete console.log(this.product)

**Review by BodyDuk — CHANGES_REQUESTED** [review]

File: UI/angular/src/app/components/cart/cart.component.html [hunk 12]
```diff
@@ -1,47 +1,45 @@
...
-  <div *ngIf="cartItems.length > 0">
-    <div class="font-size-sm mt-3">
-      <div class="row border-bottom border-dark d-none d-md-flex fw-bold">
-        <div class="col-4">Name</div>
-        <div class="col-5">Note</div>
-        <div class="col-1"><i class="bi bi-basket"></i></div>
-        <div class="col-1"><i class="bi bi-wallet"></i></div>
-        <div class="col-1"><i class="bi bi-x-lg"></i></div>
+<div class="container py-5">
+  <h3 class="mb-4">Shopping Bag ({{cartItems.length}})</h3>
```
Comment (bdemediuk) [inline UI/angular/src/app/components/cart/cart.component.html:2]: Use mt

File: UI/angular/src/app/components/cart/cart.component.html [hunk 13]
```diff
@@ -1,47 +1,45 @@
...
-        <div class="col-12 col-md-1">
-          <button class="btn d-none d-md-block"
-                  (click)="delete(i)">
-            <i class="bi bi-trash"></i>
-          </button>
-        </div>
+      <div class="col-12 col-md-1 d-flex justify-content-between mb-2">
+        <span>
+          <span class="d-md-none fw-bold">Price: </span>
+          <span>${{dataItem.quantity*getProductByProductId(dataItem.productId).price}}</span>
```
Comment (bdemediuk) [inline UI/angular/src/app/components/cart/cart.component.html:30]: Add spaces.

File: UI/angular/src/app/components/cart/cart.component.html [hunk 14]
```diff
@@ -50,13 +48,14 @@
...
-    <div class="row">
-      <div class="col-12 col-md-4">
-        <h5 class="mb-6 text-transform-none">Your shopping bag is empty.</h5>
-        <button routerLink="/"
-                class="btn btn-outline-dark my-2 px-4 py-2">Continue Browsing</button>
-      </div>
+  <div class="row"
+       *ngIf="cartItems.length === 0">
+    <div class="col-12 col-md-4">
+      <h5 class="mb-6 text-transform-none">Your shopping bag is empty.</h5>
```
Comment (bdemediuk) [inline UI/angular/src/app/components/cart/cart.component.html:54]: Why do you need `text-transform-none`

Comment (bdemediuk) [reply UI/angular/src/app/components/cart/cart.component.html:54]: Use mt

File: UI/angular/src/app/components/orders/order-collapse/order-collapse.component.html [hunk 15]
```diff
@@ -10,13 +10,13 @@
...
   <div class="col-3">${{order.totalPrice}}</div>
-  <div id="{{order.id}}"
-       *ngIf="expanded">
-    <div class="pt-4 pl-3 pr-0">
-      <div class="row">
-        <div class="col-11 offset-1">
-          <ng-content></ng-content>
-        </div>
+</div>
+<div id="{{order.id}}"
```
Comment (bdemediuk) [inline UI/angular/src/app/components/orders/order-collapse/order-collapse.component.html:14]: Why you use id="{{order.id}}"?

**Review by bdemediuk — CHANGES_REQUESTED** [review]

**Review by bdemediuk — APPROVED** [review]

**Pull request merged**